src/handlers.js

```diff
@@ -93,33 +89,20 @@ export function taskParentGen(id)
 }
 export function eventDispatcher(requestedAppId,content)
 {
+    if(requestedAppId==='editor'){
+        appInstaller('editor','./icons/notepad.png',<Editor value={content}/>)
+    }
     let id=idGen(10);
     var isRunning=isAlreadyRunning(appRegistry[requestedAppId].processName)
     //may add a if statement here 
-    if(requestedAppId==='editor'){
-        console.log(content)
-        eventHandler[id] = new runningTasks(id,appRegistry[requestedAppId],content)
-        if(!isRunning)
-        {
-            var tPG=taskParentGen(id);
-            console.log('here')
-            ReactDOM.render(<TaskbarIcon id={id} name={eventHandler[id].processName} location={`${eventHandler[id].iconData}`}/>,tPG)
-        }
-        ReactDOM.render(<Window id={id} source={eventHandler[id].Source}/>,parentGen(id));
-
-    }else{
         eventHandler[id]=new runningTasks(id,appRegistry[requestedAppId]);
         console.log(eventHandler[id]);
             if(!isRunning)
         {
             var tPG=taskParentGen(id);
-            console.log('here')
             ReactDOM.render(<TaskbarIcon id={id} name={eventHandler[id].processName} location={`${eventHandler[id].iconData}`}/>,tPG)
         }
-        console.log('here')
         ReactDOM.render(<Window id={id} source={eventHandler[id].Source}/>,parentGen(id));
```

Comment:
I meant the data should be passed here to the window component.

Reply:
Can you explain me what the window component does ? How does it render the editor component inside it ? If I pass a content component how does it act as a prop to the editor? The source that the window component has here , refers to the initial editor when installed by the app. How will it rewrite the components there?

Reply:
So yeah, I figured out a workaround... Let me explain the current model i.e Window is generic window container for holding our apps. If you look into https://github.com/NJACKWinterOfCode/JS-OS/blob/3876a1d5887d0bc898074dbd729864119714132a/src/window/window.js#L95 
this line you can see react is cloning element from the source component i.e editor in this case and more interesting thing to note is i am passing id of window(which was passed during event dispatch) to the child component as a prop. Similarly we can do something here to make it work.

First in event Dispatcher 
```suggestion
        ReactDOM.render(<Window id={id} source={eventHandler[id].Source} data={content}/>,parentGen(id));
```

Then go over to window .js
```suggestion
        React.cloneElement(this.props.source,
                    {id:this.state.id,data:this.props.data})
```

Then onto editor.js
```suggestion
              code: this.props.data,
```


It works for my test data locally atleast and it all makes sense to do this way. So please try and let me know. Thanks!

Reply:
I didnt know how suggestions in github were working so i messed it up lol, But I believe you got my point.

Reply:
Damn, I just figured that out too. 😄 I'm currently implementing this. Let's see if it works.